kura/org.eclipse.kura.web2/src/main/java/org/eclipse/kura/web/server/GwtNetworkServiceImpl.java

```diff
@@ -989,34 +988,32 @@ public void updateNetInterfaceConfigurations(GwtXSRFToken xsrfToken, GwtNetInter
                 }
             }
 
-            return new ArrayList<GwtFirewallOpenPortEntry>(gwtOpenPortEntries);
+            return new ArrayList<>(gwtOpenPortEntries);
 
         } catch (KuraException e) {
             throw new GwtKuraException(GwtKuraErrorCode.INTERNAL_ERROR, e);
         }
     }
 
     @Override
-    public ArrayList<GwtWifiHotspotEntry> findWifiHotspots(GwtXSRFToken xsrfToken, String interfaceName)
-            throws GwtKuraException {
+    public ArrayList<GwtWifiHotspotEntry> findWifiHotspots(GwtXSRFToken xsrfToken, String interfaceName,
+            String wirelessSsid) throws GwtKuraException {
 
         checkXSRFToken(xsrfToken);
         NetworkAdminService nas = ServiceLocator.getInstance().getService(NetworkAdminService.class);
         SystemService systemService = ServiceLocator.getInstance().getService(SystemService.class);
-        List<GwtWifiHotspotEntry> gwtWifiHotspotsEntries = new ArrayList<GwtWifiHotspotEntry>();
+        List<GwtWifiHotspotEntry> gwtWifiHotspotsEntries = new ArrayList<>();
 
         try {
-            Map<String, WifiHotspotInfo> wifiHotspotInfoMap = nas.getWifiHotspots(interfaceName);
-            if (wifiHotspotInfoMap != null && !wifiHotspotInfoMap.isEmpty()) {
-                Collection<WifiHotspotInfo> wifiHotspotInfoCollection = wifiHotspotInfoMap.values();
-                Iterator<WifiHotspotInfo> it = wifiHotspotInfoCollection.iterator();
-                while (it.hasNext()) {
-                    WifiHotspotInfo wifiHotspotInfo = it.next();
+            List<WifiHotspotInfo> wifiHotspotInfoList = nas.getWifiHotspotList(interfaceName);
+            if (wifiHotspotInfoList != null && !wifiHotspotInfoList.isEmpty()) {
```

Comment:
Is this check `&& !wifiHotspotInfoList.isEmpty()` needed?
I think that the for loop manages this case.